src/Illuminate/Routing/ResourceRegistrar.php

```diff
@@ -228,10 +253,49 @@ protected function getGroupResourceName($prefix, $resource, $method)
      */
     public function getResourceWildcard($value)
     {
+        if (isset($this->wildcards[$value])) {
+            $value = $this->wildcards[$value];
+        } elseif (isset(static::$wildcardMap[$value])) {
+            $value = static::$wildcardMap[$value];
+        } elseif ($this->wildcards === 'singular' || static::$singularWildcards) {
+            $value = Str::singular($value);
+        }
+
         return str_replace('-', '_', $value);
     }
 
     /**
+     * Set the global wildcard mapping.
+     *
+     * @param array $wildcards
+     * @return void
+     */
+    public static function setWildcards(array $wildcards = [])
```

Comment:
I thought taylor agreed on slack that users shouldn't need to know about the ResourceRegistrar and act on the main router class? Also this should be added in the contract for the ResourceRegistrar.